Switching back to httpOnly=true as the default

kgn opened the pull request:
After a discussion about the security implications of switching the default to httpOnly=true on the #deveopment slack channel, here's a PR to switch it back. I figured if httpOnly should be the default, it should be the default on the `Cookie init`.

0xTim commented on Sources/Auth/Authentication/AuthMiddleware.swift:
```diff
@@ -26,9 +26,7 @@ public class AuthMiddleware<U: User>: Middleware {
-                expires: Date().addingTimeInterval(cookieTimeout),
-                secure: false,
-                httpOnly: false
+                expires: Date().addingTimeInterval(cookieTimeout)
```
I would personally still set `httpOnly: true` in here to be explicit and ensure we have a HTTP only cookie for the AuthMiddleware. I would add another parameter to the `init` of `AuthMiddleware` to allow cookies to be set to `secure` but default it to `false`.
In an ideal world I would hardcod this to true but I realise that this isn't feasible and I've probably been wearing my tin-foil hat too long 😝

kgn commented:
I changed `httpOnly` back to `true` in the _cookie factory_, and `httpOnly` back to `false` in the _Cookie_ `init`. This this brings it back to exactly how the defaults were before 1.5.6, but with the additional unit tests I added.

tannernelson commented:
I replaced `1.5.6` with this now since it was such a minor change and the previous was a vulnerability. Thanks for fixing this @kgn. At least we got some additional tests out of it.

If you want to submit you PR to pass value, name for Vapor 2 that seems like the best solution now.
